sbncode/FluxReader/DK2NuInterface.cxx

```diff
@@ -0,0 +1,291 @@
+#include "DK2NuInterface.h"
+#include "dk2nu/tree/dk2nu.h"
+#include "dk2nu/tree/dkmeta.h"
+#include "dk2nu/tree/NuChoice.h"
+#include "dk2nu/tree/calcLocationWeights.h"
+
+#include "TFile.h"
+#include "TTree.h"
+
+#include <iomanip>
+
+namespace fluxr {
+  DK2NuInterface::DK2NuInterface()
+  {
+  }
+
+  DK2NuInterface::~DK2NuInterface()
+  {
+  }
+
+  void DK2NuInterface::SetRootFile(TFile* rootFile)
+  {
+    fDk2NuTree=dynamic_cast<TTree*>(rootFile->Get("dk2nuTree"));
+    fDkMetaTree=dynamic_cast<TTree*>(rootFile->Get("dkmetaTree"));
+
+    fDk2Nu = new bsim::Dk2Nu;
+    fDkMeta= new bsim::DkMeta;
+    fNuChoice=new bsim::NuChoice;
+    fDk2NuTree->SetBranchAddress("dk2nu",&fDk2Nu);
+    fDkMetaTree->SetBranchAddress("dkmeta",&fDkMeta);
+
+    fNEntries=fDk2NuTree->GetEntries();
+    fDkMetaTree->GetEntry(0);
+    fRun=fDkMeta->job;
+    fPOT=fDkMeta->pots;
+  }
+
+  void DK2NuInterface::Init(fhicl::ParameterSet const & ps)
+  {
+    //code to handle flux window stolen from GENIE_R21210/src/FluxDrivers/GNuMIFlux.cxx
+    //rotation matrix
+    try {
+      std::vector<double> rotmat=ps.get<std::vector<double> >("rotmatrix");
+      TVector3 newX,newY,newZ;
+      if (rotmat.size()==9) {
+	std::cout<<"Matrix defined with new axis values"<<std::endl;
```

Comment:
In general, I think all printouts should be prefixed with the name of the file that printed them, at least. Otherwise some of these are quite cryptic, and it will be very difficult to track down what they actually mean. Some people would insist on using the proper message facility.